File: 0155-min-stack/0155-min-stack.java
class MinStack {
    private int[] stack;
    private int[] minStack;
    private int size;
    public MinStack() {
        stack = new int[16];
        minStack = new int[16];
        size = 0;
    }
    public void push(int value) {
        if (size == stack.length) {
            stack = java.util.Arrays.copyOf(stack, size * 2);
            minStack = java.util.Arrays.copyOf(minStack, size * 2);
        }
        stack[size] = value;
        minStack[size] = (size == 0) ? value : Math.min(value, minStack[size - 1]);
        size++;
    }

    public void pop() {
        size--;
    }

    public int top() {
        return stack[size - 1];
    }

    public int getMin() {
        return minStack[size - 1];
    }
}

File: 0155-min-stack/0155-min-stack.py
class MinStack:

    def __init__(self):
        self.stack=[]
        self.minstack=[]
    def push(self, value: int) -> None:
        self.stack.append(value)
        self.minstack.append(value)
        self.minstack.sort(reverse=True)
    def pop(self) -> None:
        x = self.stack.pop(-1)
        self.minstack.remove(x)
    def top(self) -> int:
        return self.stack[-1]
    def getMin(self) -> int:
        return self.minstack[-1]


# Your MinStack object will be instantiated and called as such:
# obj = MinStack()
# obj.push(value)
# obj.pop()
# param_3 = obj.top()
# param_4 = obj.getMin()